Sources/SwiftCompilerPlugin/CompilerPlugin.swift

```diff
@@ -242,3 +244,10 @@ private extension FileHandle {
     }
   }
 }
+
+struct CompilerPluginError: Error, CustomStringConvertible {
+  var description: String
+  init(message: String) {
+    self.description = message
+  }
+}
```

Comment:
Do you think there would be value in structuring the errors and making `CompilerPluginError` an `enum` with a computed description property instead of shoving an entire error message in there?

Reply:
Not at this point. As long as we only need the message, I don't think make it an enum benefits anything.

Reply:
I was also considering commenting on this. It would be nice if rather than passing a random string (which we've duplicated in many places) into the eventual diagnostic, that we instead passed the paths/macro type/plugin kind along. I'm not sure how difficult that would be or if it's even feasible though so maybe it's not worth it.

As an example, one of the messages in this PR is `macro implementation type '\(moduleName).\(typeName)' could not be found in executable plugin '\(pluginPath)'` and another has just `type` instead. And IMO I think it would be better with neither.

Reply:
I have a plan to consolidate logics of plugin-server and in-process in the follow-ups (because they are both library plugins). In that change, I think we can consolidate this error message too between all three of executable, server, and in-process.

Reply:
It’s not a blocker for me. It was just something that crossed my mind but I’ll leave it up to you to decide how to model the error.

Reply:
Okay, that would be nice 👍